src/evaluator/index.js

```diff
@@ -256,16 +256,18 @@ export class Evaluator {
           // For reference: this is the problematic bit in web3.js:
           // https://github.com/ethereum/web3.js/blob/7d1b0eab31ff6b52c170dedc172decebea0a0217/packages/web3-eth-abi/src/index.js#L110
           name: 'nonEmptyName'
-        })),
-        outputs
-      }, inputs.map((input) => input.value))
+        }))
+      }, inputs.map((input) => input.value.toString()))
```

Comment:
Note that I added a `.toString()` since I had some issues with `BN` being passed to Web3 which would give me some weird encoding errors. I argue that this is safe because `BN` has an official `toString` implementation, as does every JavaScript type. Going by the Web3.js documentation, this function also expects every argument to be a string(?).

Reply:
> I had some issues with BN being passed to Web3 which would give me some weird encoding errors

Web3.js starting from `1.0.0-beta.36` switched to ethers.js' ABICoder implementation, and this particular behaviour was noted in https://github.com/ethers-io/ethers.js/issues/595. Given the discussion there, we should stick to using strings.

I'd have to double check, but I think all other `input.values` are strings anyway though, no?

Reply:
Yes, that is the case as far as I can tell.